김현국/app/src/main/java/com/example/myapplication/presenter/view/detail/DetailFragment.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.myapplication.base.BaseFragment
+import com.example.myapplication.databinding.FragmentDetailBinding
+import com.example.myapplication.presenter.UiState
+import com.example.myapplication.presenter.adapter.DetailUserFollowerAdapter
+import com.example.myapplication.presenter.adapter.DetailUserFollowingAdapter
+import com.example.myapplication.servicelocator.RepositoryServiceLocator
+import com.example.myapplication.servicelocator.UseCaseServiceLocator
+import com.example.myapplication.util.loadImage
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class DetailFragment : BaseFragment<FragmentDetailBinding>(FragmentDetailBinding::inflate) {
+    private val args by navArgs<DetailFragmentArgs>()
+    private val repo by lazy { args.presenterRepository }
+    private val detailUserFollowingAdapter: DetailUserFollowingAdapter by lazy { DetailUserFollowingAdapter() }
+    private val detailUserFollowerAdapter: DetailUserFollowerAdapter by lazy { DetailUserFollowerAdapter() }
+    private val detailUserRepository by lazy { RepositoryServiceLocator.provideDetailUserRepository() }
+    private val model: DetailViewModel by viewModels {
+        DetailViewModelFactory(
+            this,
+            UseCaseServiceLocator.provideGetUserFollowerUseCase(detailUserRepository = detailUserRepository),
+            UseCaseServiceLocator.provideGetUserFollowingUseCase(detailUserRepository = detailUserRepository)
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setListener()
+        initObserve()
+        initUI()
+    }
+
+    private fun initUI() = with(binding) {
+        rvDetailUserFollower.layoutManager =
+            LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
+        rvDetailUserFollower.adapter = detailUserFollowerAdapter
+
+        rvDetailUserFollowing.layoutManager =
+            LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
+        rvDetailUserFollowing.adapter = detailUserFollowingAdapter
+
+        ivDetailImage.loadImage(repo.owner.image)
+        tvDetailRepoName.text = repo.name
+        tvDetailStars.text =
+            StringBuilder().append("\u2605").append(repo.stars).append(" stars").toString()
+        tvDetailDescriptionSmall.text = repo.description
+        tvDetailLanguageSmall.text = repo.language
+        tvDetailLastUpdateSmall.text = repo.lastUpdated
+        tvDetailUserName.text = repo.owner.login
+
+        with(model) {
+            if (!savedStateHandle.contains(USER_NAME)) {
+                getUserFollowing(repo.owner.login)
+                getUserFollower(repo.owner.login)
+            }
+        }
+    }
+
+    private fun initObserve() = with(viewLifecycleOwner.lifecycleScope) {
+        launch {
```

Comment:
함수에 with(model) 해주면 한 depth가 줄어들 것 같아요.
```suggestion
    private fun initObserve() = with(viewLifecycleOwner.lifecycleScope) = with(model) {
        launch {
```

Reply:
민욱님 이렇게 한줄 선언이 가능한건가요???
LifecycleCoroutineScope를 인식하지 못하는거같아요

Reply:
아!!! ㅋㅋㅋㅋㅋㅋ 제가 실수 했네요.. ㅋㅋㅋㅋㅋㅋ 이미 with으로 감싸져 있었군요..

Reply:
바꾼다면 
`private fun test() = with(viewLifecycleOwner.lifecycleScope) {   
        with(model) {   
            launch {   `
이렇게 model까지 with으로 위에서 처리하면 될 것 같네영